providers/base/bin/reboot_check_test.sh

```diff
@@ -0,0 +1,91 @@
+#!/bin/bash
+
+device_check="true"
+service_check="true"
+
+service_check() {
+    echo -e "\n## Check system service.."
+    COUNT=$(systemctl --system --no-ask-password --no-pager --no-legend list-units --state=failed | wc -l)
+    printf "Found %s failed units\n" "$COUNT"
+    if [ "$COUNT" != 0 ]; then
+        printf "\nFailed units:\n"
+        systemctl --system --no-ask-password --no-pager list-units --state=failed
+        service_check="false"
+    fi
+}
+
+dump() {
+    echo "## Dump the devices information to $output_dir.."
+    mkdir -p "$output_dir"
+    lspci -i "$SNAP"/usr/share/misc/pci.ids > "$output_dir"/lspci_log
+    iw dev | grep "Interface\|addr\|ssid" > "$output_dir"/wifi_conn
```

Comment:
I saw that originally, the job was using `nmcli`, but here you are using `iw`. Is it because `iw` is available on both Classic and Ubuntu Core? (I quickly checked on my 22.04 desktop and `iw` is not installed by default)

Reply:
the reason to using `iw` command instead of `nmcli` is because of the network-manager package is not a built-in package in the IoT server image and IoT core image. And that means the checkbox not able to get wireless interface during the stress test if the NetworkManager package is not install manually on a system running IoT server image.

So I looked into the checkbox snapcraft file and debian control file, I found the iw package is integrated into checkbox snap and debian package.
https://github.com/canonical/checkbox/blob/main/checkbox-core-snap/series22/snap/snapcraft.yaml#L274
https://github.com/canonical/checkbox/blob/main/providers/certification-client/debian/control#L34

BTW, there's a `wifi_interface_mode` resource job is using `iw` command to get the supported wifi interface modes as well.

Reply:
Thanks for the clarification!